src/models/auditModelGenerator.js

```diff
@@ -0,0 +1,149 @@
+// Load in our dependencies
+import { Model, DataTypes } from 'sequelize'; // eslint-disable-line import/no-import-module-exports
+import httpContext from 'express-http-context'; // eslint-disable-line import/no-import-module-exports
+import { auditLogger } from '../logger'; // eslint-disable-line import/no-import-module-exports
+
+const dmlType = ['INSERT', 'UPDATE', 'DELETE'];
+
+// const exception = () => {
+//   throw new Error(
+//     'Audit log only allows reading and inserting data,'
+//     + ' all modification and removal is not allowed.'
+//   );
+// };
+
+const tryJsonParse = (data) => {
+  let newData = data;
+  if (data) {
+    if (typeof data === 'object') {
+      Object.entries(data).forEach(([key, value]) => {
+        newData[key] = tryJsonParse(value);
+      });
+    } else if (typeof data === 'string') {
+      try {
+        newData = JSON.parse(data);
+      } catch (e) {
+        newData = data;
+      }
+    }
+  }
+  return newData;
+};
+
+const addAuditTransactionSettings = async (sequelize, instance, options, type, name) => {
+  const loggedUser = httpContext.get('loggedUser') ? httpContext.get('loggedUser') : '';
+  const transactionId = httpContext.get('transactionId') ? httpContext.get('transactionId') : '';
+  const sessionSig = httpContext.get('sessionSig') ? httpContext.get('sessionSig') : '';
+  const auditDescriptor = httpContext.get('auditDescriptor') ? httpContext.get('auditDescriptor') : '';
+  if (loggedUser !== '' || transactionId !== '' || auditDescriptor !== '') {
```

Comment:
I assume the audit functions can handle not having any of the variables below set (if this `if` is false)

Reply:
Defaults will be used for normal data flow, but this was also needed to support tests as there is no good way to set the httpContext for tests.

Reply:
Yes, the DB will use default values for anything not passed. This check is basically only useful for test cases, as the lack of a httpContext was making it not possible to  set the values manually.